main.c

```diff
@@ -1657,6 +1665,53 @@ static struct PyModuleDef gmp_module = {
 };
 
 
+#define max_alloc 8192
+static void* mem_arr[max_alloc];
+size_t idx = 0;
+
+
+static void *
+my_allocate_function(size_t size)
+{
+    if (idx > max_alloc) {
+        goto err;
+    }
+
+    void *ret = malloc(size);
+
+    if (ret == 0) {
+        goto err;
+    }
+    mem_arr[idx] = ret;
+    idx++;
+    return ret;
+err:
```

Comment:
Why do we need to release all the memory that has been allocated after the allocation fails?

Reply:
Because we won't memory leaks, isn't?

It seems, the GMP does only allocation of temporary buffers in ``mpn_*`` functions.  So, if it fails, we de-allocate all that stuff and jump to the caller, which will raise MemoryError.